### DISCUSSION TITLE
Setting a time field with a timestamp

### DISCUSSION THREAD

**Discussion by murraycollingwood** [propelorm/Propel2#1878] [Q&A]
Hello again

I'm struggling with an odd problem.

My table 'Period' has a field 'Hmend' which is of type 'time' is giving me an error when I try to store a timestamp in this field.
(PS This has been working for many years in Propel1.6)
Here is the error:
```
[2-5-22 3:46pm] [0]: Uncaught exception "Propel\Runtime\Exception\PropelException" with message "Error parsing date/time value: 31532399" in /home/sobs/vendor/propel/propel/src/Propel/Runtime/Util/PropelDateTime.php on line 139 
Context: 0 = Array ( [0] => Array ( [file] => /home/sobs/generated-classes/Base/Period.php [line] => 581 [function] => newInstance [class] => Propel\Runtime\Util\PropelDateTime [type] => :: [args] => Array ( [0] => 31532399 [1] => [2] => DateTime ) ) [1] => Array ( [file] => /home/sobs/build/ui/calendar.ajax.php [line] => 1460 [function] => setHmend [class] => Base\Period [type] => -> [args] => Array ( [0] => 31532399 ) ) [2] => Array ( [file] => /home/sobs/build/ui/components/ajax.php [line] => 76 [function] => periodSave [args] => Array ( [0] => 71358 [1] ...
```
From calendar.ajax.php near line 1460:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/5810216/166624828-ef668387-fe25-45c7-b0ac-81ee071454a6.png">

My definition of the Hmend field in the schema file:
```
  <column name="hmend" type="time" description="hh:mm:ss AM"/>
```
And in the database:
<img width="684" alt="image" src="https://user-images.githubusercontent.com/5810216/166624963-3e7e953a-82fc-4b6f-9f2a-2126bcaf0faa.png">

This is happening every time.
I'm still investigating what is causing this.

**Comment by murraycollingwood** [comment]
I think I found the problem in PropelDateTime::isTimestamp
```
        if (strlen((string)$value) === 8) {
            return false;
        }
```

Somebody decided that an 8 digit number is not a timestamp... but it is if it is just a time value (and not a date).
Not hard to kludge this time, I just started with todays date and set the time-of-day.  
However, I'm sure somebody might have a better long term solution than this.

**Comment by mringler** [comment]
Hmmm, I am not too familiar with the way Propel handles time values, but it looks like you have to use a format that php DateTime understands (https://www.php.net/manual/en/datetime.formats.time.php).
Where does your timestamp (31532399 - apparently Thursday, December 31, 1970 10:59:59 PM) come from?

**Comment by murraycollingwood** [comment]
I checked, Propel accepts a string, a DateTime object or a timestamp.
I'm storing a time value only, to an SQL "time" field.
I'm setting up times for periods, and in this instance the end time is 11pm minus 1 second.
Honestly I can't say where the 31st of December comes from.

However, specifying an 8 digit number is not a valid timestamp makes these timestamps invalid:
From: 10000000 (1970-04-26 17:46:40)
Through: 99999999 (1973-03-03 09:46:39)

Can we have that test removed please?

**Comment by mringler** [comment]
the thing with the 8 digit number seems strange, but apparently, it is there to allow for date input in format Ymd, as in '20220505'. Tests fail when removing it:
```
1) Propel\Tests\Runtime\Util\PropelDateTimeTest::testNewInstance with data set "Ymd" ('20110810', '2011-08-10 00:00:00')
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'2011-08-10 00:00:00'
+'1970-08-21 18:20:10'
```
But I have to agree, disabling a number of valid timestamps is a pretty harsh side effect. Not sure what a good solution would be.

**Comment by dereuromark** [comment]
Well, since those are as string, I would assume that as int we can allow normal date integers
What do you think about

    if (is_string($value) && strlen($value) === 8) {
    
Maybe we can even deprecate that part and require it to be a consistent documented format.
